drivers/serial/uart_emul.c

```diff
@@ -127,6 +230,15 @@ uint32_t uart_emul_put_rx_data(const struct device *dev, uint8_t *data, size_t s
 	key = k_spin_lock(&drv_data->rx_lock);
 	count = ring_buf_put(drv_data->rx_rb, data, size);
 	k_spin_unlock(&drv_data->rx_lock, key);
+
+	if (count > 0 && drv_data->irq_en) {
+
+		drv_data->irq_en = true;
```

Comment:
`drv_data->irq_en` should be already `true` if the conditionals above is `true`

Reply:
Fixed!